guides/Moonscript_Yuescript.md

```diff
@@ -0,0 +1,207 @@
+Moonscript & Yuescript
+===
+
+[Yuescript](https://yuescript.org/) is a 
+[Moonscript](https://moonscript.org) dialect.
+Like Moonscript it
+[trans-compiles](https://en.wikipedia.org/wiki/Source-to-source_compiler)
+into Lua, thus both languages are LÖVR scripting alternatives.
+
+After the setup (in conf.lua) Lua and Yuescript/Moonscript modules can be required from each other
+in the usual way.
+
+It is also possible to enable support for both dialects.
+
+Why and Which
+---
+
+Be warned, both languages are [white space sensitive](https://en.wikipedia.org/wiki/Off-side_rule).
+If you can't stand this kind of syntax stop reading here.
+
+Moonscript dialects come with a streamlined table constructor syntax,
+making them suited not only for scripting but also for storage of your game's content data.
+
+Featured are all kinds of additional syntactic sugar
+and a class system for object oriented coding.
+
+At the time of this writing (December 2022)
+Yuescript is under active development and extends on Moonscript a lot already.
+Moonscript on the other hand is more mature, more stable, features more/better editor support and is
+better documented.
+
+Precompiled Code
+---
+
+The simplest setup is compiling the codebase to Lua before shipping.
+No extra dependencies are required in this setup.
+
+### Yuescript
+
+Use the '-l' flag which enables commenting each line of the Lua file with the corresponding
+line-number in the source file:
+    
+    $ yue -l yourGame
+
+A special error handler can extract this information 
+and point to the correct line in the source file when an error is thrown.
+
+(TODO: Write more about this error handler function. I guess some Yuescript function can be reused.)
+
+### Moonscript
+
+    $ moonc yourGame
+
+(TODO: Write more about error line rewriting.)
+
+Runtime Compilation
+---
+
+The more elegant solution is the trans-compilation at runtime.
+Source and Lua files can't be out of sync and the setup is more developer friendly.
+
+### Yuescript
+
+Yuescript comes also (beside the standalone executable) in form of a Lua c library.
+You need to ship a different one for each operating system you want to support.
+
+The Yuescript compiler takes care of the line-number translation,
+no extra work needed in this setup.
+
+Setup in conf.lua:
+
+``` lua
```

Comment:
This is embarrassing, but the markdown converter on the website doesn't support fenced codeblocks (```) right now.  Best to use 4-space indentation (the JS syntax highlighter will automatically detect the language).

Reply:
Noted